pkg/ddl/backfilling_scheduler.go

```diff
@@ -501,15 +524,21 @@ func (b *ingestBackfillScheduler) createCopReqSenderPool() (*copReqSenderPool, e
 	return newCopReqSenderPool(b.ctx, copCtx, sessCtx.GetStore(), b.taskCh, b.sessPool, b.checkpointMgr), nil
 }
 
-func (*ingestBackfillScheduler) expectedWorkerSize() (readerSize int, writerSize int) {
-	return expectedIngestWorkerCnt()
+func (b *ingestBackfillScheduler) expectedWorkerSize() (readerSize int, writerSize int) {
+	return expectedIngestWorkerCnt(b.avgRowSize)
 }
 
-func expectedIngestWorkerCnt() (readerCnt, writerCnt int) {
+func expectedIngestWorkerCnt(avgRowSize int) (readerCnt, writerCnt int) {
 	workerCnt := int(variable.GetDDLReorgWorkerCounter())
-	readerCnt = min(workerCnt/2, maxBackfillWorkerSize)
-	readerCnt = max(readerCnt, 1)
-	writerCnt = min(workerCnt/2+2, maxBackfillWorkerSize)
+	readerRatio := []float64{0.5, 1, 2, 4, 8}
+	rowSize := []int{200, 500, 1000, 3000, 5000}
+	for i, s := range rowSize {
+		if avgRowSize <= s {
+			readerCnt = int(float64(workerCnt) * readerRatio[i])
+			writerCnt = workerCnt
```

Comment:
also. < 200, the writerCnt need +2 otherwise, I remeber the performance will degrade bigger than 10%

Reply:
I prefer to make it simple and easy to predict.

Reply:
OK, let's see the perf testing result.